test/storage.test.js

```diff
@@ -287,4 +288,158 @@ describe('storage.js', () => {
+                .get('/photo/elephant.png')
+                .reply(404, 'ok');
+
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                console.log(e);
+                assert.ok(e instanceof GenericError);
+                assert.equal(e.message, "GET 'https://example.com/photo/elephant.png' failed with status 404");
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should not download a file in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+
+            const params = {
+                watermarkContent: 'file.png'
+            };
+            const inDirectory = '/in';
+
+            mockFs({ '/in/file.png': 'yo' });
+            const watermark = await getWatermark(params, inDirectory);
+
+            assert.equal(watermark.name, 'watermark.png'); // in this case source name is actual file path
+            assert.equal(watermark.path, '/in/watermark.png');
+        });
+
+        it('should fail to download because path ends with /..', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'file.jpg/..'
+            };
+            const inDirectory = '/in';
+
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file file.jpg/..');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail when path contains traversal notation in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'file/../../../../evilcode/elephant.jpg'
+            };
+            const inDirectory = '/in';
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file file/../../../../evilcode/elephant.jpg');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail because of missing localfile in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'elephant.jpg'
+            };
+            const inDirectory = '/in';
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file elephant.jpg');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail because watermark asset is not png', async () => {
```

Comment:
Once you add the file-type lib based check, please add a test using https://github.com/adobe/asset-compute-sdk/blob/master/test/files/negative/file-webp-masquerading-as-png.png. This file is a webp file which improperly has a png extension (this is a type of error/mistake we actually encountered in prod envs, so I think it's justified to test it here too).

Reply:
Needs a ticket for later